rust/src/vacuum.rs

```diff
@@ -0,0 +1,217 @@
+//! Vacuum a Delta table
+//!
+//! Run the Vacuum command on the Delta Table: delete files no longer referenced by a Delta table and are older than the retention threshold.
+//! We do not recommend that you set a retention interval shorter than 7 days, because old snapshots
+//! and uncommitted files can still be in use by concurrent readers or writers to the table.
+//! If vacuum cleans up active files, concurrent readers can fail or, worse, tables can be
+//! corrupted when vacuum deletes files that have not yet been committed.
+//! If `retention_hours` is not set then the `configuration.deletedFileRetentionDuration` of
+//! delta table is used or if that's missing too, then the default value of 7 days otherwise.
```

Comment:
One other warning that may be worth calling out (even though it's implicit here): vacuuming for before a certain date means time travel may not work for commits before that date.